src/coreservices.cpp

```diff
@@ -441,7 +441,9 @@ void CoreServices::initialize(QApplication* pApp) {
     // Load tracks in args.qlMusicFiles (command line arguments) into player
     // 1 and 2:
     const QList<QString>& musicFiles = m_cmdlineArgs.getMusicFiles();
-    for (int i = 0; i < (int)m_pPlayerManager->numDecks() && i < musicFiles.count(); ++i) {
+    const int numTracks = std::min(m_pPlayerManager->numberOfDecks(),
+            static_cast<int>(musicFiles.count()));
+    for (int i = 0; i < numTracks; ++i) {
```

Comment:
What's the benefit of preincr vs postincr? Is this preferred?

Reply:
`i++` was the original one preferred in C and the origin of the name for C++. In C++ itself `++i` is preferred if possible, because it returns the incremented value and not the original one which is considered to be faster in case of complex types, because the original one does not need to be kept in memory. In this particular case the optimized machine code will most likely be the same anyway because the return value is not used.

Reply:
Thanks for the detailed explanation. I knew about the usecase where using the value, but I was wondering if there is reason why you would use this form when the value is irrelevant. Is that something we should [unify](https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines)? I know we have a few for-loop which use the two form, beside no apparent reason for switching between the two forms

Reply:
I don't really mind. Both is correct and readable in this case IMHO.